jiant/modules/seq2seq_decoder.py

```diff
@@ -148,9 +208,16 @@ def forward(
         """
         batch_size, _, _ = encoder_outputs.size()
 
-        use_gold = target_tokens is not None and self.training
+        if self.training:
+            assert target_tokens
+            mode = "train"
+        elif target_tokens is not None:
+            assert target_tokens
```

Comment:
why would this assert ever fail given the above condition ?

Reply:
True, took this out. Thanks!